nemo/collections/asr/models/regression_models.py

```diff
@@ -0,0 +1,331 @@
+
+    def setup_test_data(self, test_data_layer_params: Optional[Union[DictConfig, Dict]]):
+        if 'shuffle' not in test_data_layer_params:
+            test_data_layer_params['shuffle'] = False
+
+        self.test_manifest = test_data_layer_params.get('manifest_filepath', None)
+        self._test_dl = self.__setup_dataloader_from_config(config=test_data_layer_params)
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        if hasattr(self.preprocessor, '_sample_rate'):
+            audio_eltype = AudioSignal(freq=self.preprocessor._sample_rate)
+        else:
+            audio_eltype = AudioSignal()
+        return {
+            "input_signal": NeuralType(('B', 'T'), audio_eltype),
+            "input_signal_length": NeuralType(tuple('B'), LengthsType()),
+        }
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {"preds": NeuralType(tuple('B'), RegressionValuesType())}
+
+    @typecheck()
+    def forward(self, input_signal, input_signal_length):
+        processed_signal, processed_signal_len = self.preprocessor(
+            input_signal=input_signal, length=input_signal_length,
+        )
+
+        encoded, _ = self.encoder(audio_signal=processed_signal, length=processed_signal_len)
+        logits = self.decoder(encoder_output=encoded)
+        return logits.view(-1)
+
+    # PTL-specific methods
+    def training_step(self, batch, batch_idx):
+        audio_signal, audio_signal_len, targets = batch
+        logits = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        loss = self.loss(preds=logits, labels=targets)
+
+        self.log('loss', loss)
+        self.log('learning_rate', self._optimizer.param_groups[0]['lr'])
+
+        return {'loss': loss}
+
+    def validation_step(self, batch, batch_idx, dataloader_idx: int = 0):
+        audio_signal, audio_signal_len, targets = batch
+        logits = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        loss_value = self.loss(preds=logits, labels=targets)
+
+        return {
+            'val_loss': loss_value,
+        }
+
+    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
+        val_loss_mean = torch.stack([x['val_loss'] for x in outputs]).mean()
+
+        logging.info("val_loss: {:.3f}".format(val_loss_mean))
+        self.log('val_loss', val_loss_mean)
+
+        return {
+            'val_loss': val_loss_mean,
+        }
+
+    def test_step(self, batch, batch_idx, dataloader_idx: int = 0):
+        audio_signal, audio_signal_len, targets = batch
+        logits = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        loss_value = self.loss(preds=logits, labels=targets)
+
+        return {
+            'test_loss': loss_value,
+        }
+
+    def multi_test_epoch_end(self, outputs, dataloader_idx: int = 0):
```

Comment:
For multi methods, please return a log dict. It manages the actual logging with self.log() internally while properly attaching dataset prefix names.